src/transactions/transactions.util.ts

```diff
@@ -53,11 +70,42 @@ export async function processTransaction<
 >(
   method: Method<MethodArgs, ReturnType, TransformedReturnType>,
   args: MethodArgs,
-  opts: ProcedureOpts
+  opts: ProcedureOpts,
+  dryRun = false
 ): Promise<TransactionResult<TransformedReturnType>> {
   try {
     const procedure = await prepareProcedure(method, args, opts);
-    const result = await procedure.run();
+
+    const batch: [
+      feesPromise: Promise<PayingAccountFees>,
+      resultPromise: Promise<TransformedReturnType | undefined>
+    ] = [procedure.getTotalFees(), dryRun ? Promise.resolve(undefined) : procedure.run()];
+
+    const supportsSubsidy = procedure.supportsSubsidy();
+    const [totalFees, result] = await Promise.all<
+      PayingAccountFees,
+      TransformedReturnType | undefined
+    >(batch);
+
+    const {
+      fees,
+      payingAccountData: { balance, type, account },
+    } = totalFees;
+
+    const details: TransactionDetails = {
+      status: procedure.status,
+      fees,
+      supportsSubsidy,
+      payingAccount: {
+        balance,
+        type,
+        address: account.address,
+      },
+    };
+
+    if (dryRun) {
+      return { details, result: result as unknown as TransformedReturnType, transactions: [] };
```

Comment:
In production code you should never cast to `unknown` and then to a type

If you needed to do this for some reason the runtime logic should be encapsulated in a type guard: https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates